packages/musap-rn-kms/src/agent/MusapKeyManagerSystem.ts

```diff
@@ -0,0 +1,131 @@
+    try {
+      const keysJson: MusapKey[] = (await this.musapKeyStore.listKeys()) as MusapKey[];
+      return keysJson.map((key) => this.asMusapKeyInfo(key));
+    } catch (error) {
+      console.error('Failed to list keys:', error);
+      throw error;
+    }
+  }
+
+  async createKey(args: { type: TKeyType; sscdType?: SscdType }): Promise<ManagedKeyInfo> {
+    const keyAlgorithm = this.mapKeyTypeToAlgorithmType(args.type);
+
+    const keyGenReq: KeyGenReq = {
+      keyAlgorithm: keyAlgorithm,
+      did: '',
+      keyUsage: 'sign',
+      keyAlias: uuid(),
+      attributes: [
+        { name: 'purpose', value: 'encrypt' },
+        { name: 'purpose', value: 'decrypt' }
+      ],
+      role: 'administrator'
+    };
+
+    try {
+      const generatedKeyUri = await this.musapKeyStore.generateKey(this.sscdType, keyGenReq);
+      if (generatedKeyUri) {
+        console.log('Generated key:', generatedKeyUri);
+        const key = await this.musapKeyStore.getKeyByUri(generatedKeyUri);
+        return this.asMusapKeyInfo(key);
+      } else {
+        console.log('Failed to generate key');
+        throw new Error('Failed to generate key');
+      }
+    } catch (error) {
+      console.error('An error occurred:', error);
+      throw error;
+    }
+  }
+
+  mapKeyTypeToAlgorithmType = (type: TKeyType): KeyAlgorithmType => {
+    switch (type) {
+      case 'Secp256k1':
+        return 'ECCP256K1';
+      case 'Secp256r1':
+        return 'ECCP256R1';
+      case 'RSA':
+        return 'RSA2K';
+      default:
+        throw new Error(`Key type ${type} is not supported by MUSAP`);
+    }
+  }
+
+  async deleteKey({ kid }: { kid: string }): Promise<boolean> {
+    try {
+      await this.musapKeyStore.removeKey(kid);
+      return true;
+    } catch (error) {
+      console.error('Failed to delete key:', error);
+      return false;
+    }
+  }
+
+  async sign(args: { keyRef: Pick<IKey, 'kid'>; algorithm?: string; data: Uint8Array; [x: string]: any }): Promise<string> {
+      if (!args.keyRef) {
+        throw new Error('key_not_found: No key ref provided');
+      }
+
+      let data = ''
+      try {
+        data = new TextDecoder().decode(args.data as Uint8Array)
+      } catch (e) {
+        console.log('error on decoding the Uint8Array data', e);
+      }
+
+      const key: MusapKey = (this.musapKeyStore.getKeyById(args.keyRef.kid)) as MusapKey;
+      const signatureReq: SignatureReq = {
+        keyUri: key.keyUri,
+        data,
+        algorithm: args.algorithm as SignatureAlgorithmType ?? signatureAlgorithmFromKeyAlgorithm(key.algorithm),
+        displayText: args.displayText,
+        transId: args.transId,
+        format: args.format as SignatureFormat ?? 'RAW',
+        attributes: args.attributes
+      }
+      return this.musapKeyStore.sign(signatureReq)
+  }
+
+  async importKey(args: Omit<MinimalImportableKey, 'kms'> & { privateKeyPEM?: string }): Promise<ManagedKeyInfo> {
+    throw new Error('Not implemented.');
```

Comment:
Suggested message
"Not supported; MUSAP is a hardware  key-store which cannot import keys."

Reply:
changed to this message